registry_no_code_module.go

```diff
@@ -0,0 +1,284 @@
+// Enterprise API supports.
+//
+// TFE API docs: https://www.terraform.io/docs/cloud/api/modules.html (TODO: update this link)
+type RegistryNoCodeModules interface {
+
+	// Create a registry no-code module
+	// **Note: This API is still in BETA and subject to change.**
+	Create(ctx context.Context, organization string, options RegistryNoCodeModuleCreateOptions) (*RegistryNoCodeModule, error)
+
+	// Read a registryno-code  module
+	// **Note: This API is still in BETA and subject to change.**
+	Read(ctx context.Context, noCodeModuleID string, options *RegistryNoCodeModuleReadOptions) (*RegistryNoCodeModule, error)
+
+	// Update a registry no-code module
+	// **Note: This API is still in BETA and subject to change.**
+	Update(ctx context.Context, noCodeModuleID string, options RegistryNoCodeModuleUpdateOptions) (*RegistryNoCodeModule, error)
+
+	// Delete a registry no-code module
+	// **Note: This API is still in BETA and subject to change.**
+	Delete(ctx context.Context, ID string) error
+}
+
+// registryNoCodeModules implements RegistryNoCodeModules.
+type registryNoCodeModules struct {
+	client *Client
+}
+
+// RegistryNoCodeModule represents a registry no-code module
+type RegistryNoCodeModule struct {
+	ID                  string `jsonapi:"primary,no-code-modules"`
+	Enabled             bool   `jsonapi:"attr,enabled"`
+	FollowLatestVersion bool   `jsonapi:"attr,follow-latest-version"`
+	VersionPin          string `jsonapi:"attr,version-pin,omitempty"`
+
+	// Relations
+	Organization    *Organization           `jsonapi:"relation,organization"`
+	RegistryModule  *RegistryModule         `jsonapi:"relation,registry-module"`
+	VariableOptions []*NoCodeVariableOption `jsonapi:"relation,variable-options"`
+}
+
+// NoCodeVariableOption represents a registry no-code module variable and its
+// options.
+type NoCodeVariableOption struct {
+	// Type is a public field utilized by JSON:API to
+	// set the resource type via the field tag.
+	Type string `jsonapi:"primary,variable-options"`
+
+	// Required: The variable name
+	VariableName string `jsonapi:"attr,variable-name"`
+
+	// Required: The variable type
+	VariableType string `jsonapi:"attr,variable-type"`
+
+	// Optional: The options for the variable
+	Options []string `jsonapi:"attr,options"`
+}
+
+// RegistryNoCodeModuleCreateOptions is used when creating a registry no-code module
+type RegistryNoCodeModuleCreateOptions struct {
+	// Type is a public field utilized by JSON:API to
+	// set the resource type via the field tag.
+	// It is not a user-defined value and does not need to be set.
+	// https://jsonapi.org/format/#crud-creating
+	Type string `jsonapi:"primary,no-code-modules"`
+
+	// Required: whether the no-code module should follow the latest registry module version
+	FollowLatestVersion *bool `jsonapi:"attr,follow-latest-version"`
+
+	// Required whether no-code is enabled for the registry module
+	Enabled *bool `jsonapi:"attr,enabled"`
+
+	// Required: the registry module ID
```

Comment:
remove ID since the whole registry module object is passed:
```suggestion
	// Required: the registry module to use for the no-code module
```